### PR TITLE
Ensure repo deletion also removes associated content and artifacts

### PR DESCRIPTION
**Pull request by ShaiahWren**
Issue: AAH-2024

Prior to this pr, the `delete_container_distribution` method deleted the repo from the system, yet the associated content and artifacts remained. 

To ensure all content is deleted, the content_pks in orphan cleanup for the method is now set to the content pks for the given repository and the orphan protection time, set to 0. 

Plan of action:
1. Grab the repo ids through list comprehension of the instance_ids passed into the deletion method
2. Query objects in the RepositoryContent model and filter out the content pks associated with those repo ids. ( RepositoryContent.objects.filter(repository__pk__in = repo ids).values_list("content_pks"). 
3. Assign that list of content pks to the content pks field for orphan cleanup and set orphan_protection_time to 0. 

That should be it!

### CODE REVIEW

File: galaxy_ng/app/tasks/deletion.py [hunk 1]
```diff
@@ -34,8 +35,13 @@ def delete_container_distribution(instance_ids):
     log.info("Running core.general_multi_delete to delete distro and repo")
     general_multi_delete(instance_ids=instance_ids)
 
+    repository_ids = [item[0] for item in instance_ids]
+    repository_content_pks = \
```
Comment (newswangerd) [inline galaxy_ng/app/tasks/deletion.py:39]: This isn't working and I can't figure out why. When I add `print(repository_content_pks)`, I get `<QuerySet []>` which is wrong. It shouldn't be empty.

File: galaxy_ng/app/tasks/deletion.py [hunk 2]
```diff
@@ -34,8 +35,13 @@ def delete_container_distribution(instance_ids):
...
+        RepositoryContent.objects.filter(repository__pk__in=repository_ids)\
```
Comment (newswangerd) [inline galaxy_ng/app/tasks/deletion.py:40]: This also needs to exclude content that is in other repositories. I _think_ this should work, but you'll need to test it.

```
RepositoryContent.objects.annotate(num_repos=Count("repository")).filter(repository__pk__in=repository_ids, num_repos=1)
```

File: galaxy_ng/tests/integration/api/test_container_delete.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    )
+
+    # Check content before deleting
+    assert len(content_list["results"]) > 0
+
+    # Delete repository, contents, and artifacts
+    delete_response = client(f"{api_prefix}/v3/"
+                             "plugin/execution-environments/repositories/alpine/", method='DELETE')
+    task = monitor_task(delete_response)
+    assert len(task.created_resources) == 1
```
Comment (newswangerd) [inline galaxy_ng/tests/integration/api/test_container_delete.py:56]: why are you checking that created resource is 1?

**Review by newswangerd — CHANGES_REQUESTED** [review]
Your tests are still failing.

```
         # Delete repository, contents, and artifacts for alpine1, NOT alpine2
        delete_response = client(f"{api_prefix}/v3/"
                                 "plugin/execution-environments/repositories/alpine1/", method='DELETE')
        resp = wait_for_task(client, delete_response, timeout=10000)
        assert resp["state"] == "completed"
    
        # Ensure content hrefs from alpine1 still exists
        content_hrefs = [item["pulp_href"] for item in content_list_1["results"]]
    
        for item in content_hrefs:
            success = None
            try:
                client(f"{api_prefix}{item}")
                success = True
            except GalaxyError as ge:
                if ge.http_code in [403, 404]:
                    success = False
                else:
                    raise Exception(ge)
    
>           assert success
E           assert False
```

**Comment by ShaiahWren** [comment]
@newswangerd the first test does pass according to github, it's the second that doesn't - and from the output on github it seems like an issue with push to registry. Why is there a line that says "assert False" at the bottom of the file... is that an output of the test ran locally? (None of my container tests run locally, properly...)

```The push refers to repository [localhost:5001/alpine1]
7cd52847ad77: Preparing
7cd52847ad77: Pushed
latest: digest: sha256:e2e16842c9b54d985bf1ef9242a313f36b856181f188de21313820e177002501 size: 528
The push refers to repository [localhost:5001/alpine2]
7cd52847ad77: Preparing
7cd52847ad77: Mounted from alpine1
latest: digest: sha[256](https://github.com/ansible/galaxy_ng/actions/runs/4166837580/jobs/7211678024#step:14:257):e2e16842c9b54d985bf1ef9242a313f36b856181f188de21313820e177002501 size: 528
FAILED
'''

**Comment by newswangerd** [comment]
@ShaiahWren it says `assert False` because `False` is the value of the `success` variable

File: galaxy_ng/tests/integration/api/test_container_delete.py [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+    )
+    distro_response2 = client(
+        f"{api_prefix}/pulp/api/v3/distributions/"
+        "container/container/?base_path=alpine2",
+    )
+    assert distro_response1["results"][0]["base_path"] == 'alpine1'
+    assert distro_response2["results"][0]["base_path"] == 'alpine2'
+
+    # Grab the repository href from the json and make get request
+    repo_href_1 = (distro_response1["results"][0]["repository"]).replace("/api/automation-hub", "")
```
Comment (newswangerd) [inline galaxy_ng/tests/integration/api/test_container_delete.py:113]: You shouldn't have to replace `/api/automation-hub`. The href it gives you will work by itself

**Comment by jerabekjiri** [comment]
@ShaiahWren if you rebase, this `test_openapi.py::test_galaxy_openapi_validation`  should get fixed

File: galaxy_ng/tests/integration/api/test_container_delete.py [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+    resp = wait_for_task(client, delete_response, timeout=10000)
+    assert resp["state"] == "completed"
+
+    # Ensure content hrefs from alpine1 still exists
+    content_hrefs = [item["pulp_href"] for item in content_list_1["results"]]
+
+    for item in content_hrefs:
+        success = None
+        try:
+            client(f"{api_prefix}{item}")
```
Comment (jerabekjiri) [inline galaxy_ng/tests/integration/api/test_container_delete.py:147]: `item` is `pulp_href` and already contains full path to the content (`/api/automation-hub/pulp/api/v3/content/container/tags/9723d20f-9095-4f0c-9b5e-bcbafe387cfc/`)
```suggestion
            client(item)
```

**Review by newswangerd — APPROVED** [review]

**Pull request merged**